pkg/kn/commands/source/ping/flags.go

```diff
@@ -42,6 +43,8 @@ func (c *pingUpdateFlags) addFlags(cmd *cobra.Command) {
 
 	cmd.Flags().StringVarP(&c.data, "data", "d", "", "Json data to send")
 
+	cmd.Flags().StringVarP(&c.encoding, "encoding", "e", "", "Preferred encoding format. Valid values: text/base64")
```

Comment:
I'd suggest to change the description a bit. `Data encoding format. One of: text|base64.`

Reply:
In addition it should be also mentioned that `--data` is smarter now and detects base64 encoding.